src/core/provider.cc

```diff
@@ -762,6 +756,8 @@ InferResponseProvider::AllocateOutputBuffer(
 
   void* buffer = nullptr;
   void* buffer_userp = nullptr;
+  TRTSERVER_Memory_Type raw_actual_memory_type = preferred_memory_type;
```

Comment:
Don't set these... it is confusing and they are output values that must be set by alloc_fn_.